### PR TITLE
Cleanup System.Reflection.TypeExtensions.MemberInfo tests

### PR DESCRIPTION
**Pull request by hughbe**
- Move tests around to the place they actually belong
- Make tests do what they're actually testing rather than dodgy copy-pasting in the old tests

### CODE REVIEW

File: src/System.Reflection.TypeExtensions/tests/MemberInfo/MemberInfoMemberType.cs [hunk 1]
```diff
@@ -1,214 +0,0 @@
...
-            Type[] parameterTypes = { typeof(int) };
-            constructorInfo = typeof(TestClass1).GetConstructor(parameterTypes);
-            memberInfo = constructorInfo as MemberInfo;
-            actualValue = memberInfo.Name.Equals(".ctor"); ;
-            Assert.Equal(expectedValue, actualValue);
-        }
-
-        // PosTest6: private instance field
-        [Fact]
-        public void PosTest6()
```
Comment (stephentoub) [inline src/System.Reflection.TypeExtensions/tests/MemberInfo/MemberInfoMemberType.cs]: Where did the tests for private instance and static fields go?

File: src/System.Reflection.TypeExtensions/tests/MemberInfo/MemberInfoMemberType.cs [hunk 2]
```diff
@@ -1,214 +0,0 @@
...
-        // PosTest9: nested type
-        [Fact]
-        public void PosTest9()
-        {
-            bool expectedValue = true;
-            bool actualValue = false;
-            Type testType;
-            Type nestedType;
-            testType = typeof(ReflectionMemberInfoMemberType);
-            nestedType = testType.GetNestedType("TestClass1", BindingFlags.NonPublic);
```
Comment (stephentoub) [inline src/System.Reflection.TypeExtensions/tests/MemberInfo/MemberInfoMemberType.cs]: Where did the nested type test go?

File: src/System.Reflection.TypeExtensions/tests/MemberInfoTests.cs [hunk 3]
```diff
@@ -6,12 +6,12 @@
 
 namespace System.Reflection.TypeExtensions.Tests.MemberInfo
 {
-    public class MetadataTokenTests
+    public class MethodInfoTests
```
Comment (stephentoub) [inline src/System.Reflection.TypeExtensions/tests/MemberInfoTests.cs]: The file is called MemberInfoTests.cs and the class name is MethodInfoTests?

File: src/System.Reflection.TypeExtensions/tests/MemberInfoTests.cs [hunk 4]
```diff
@@ -6,12 +6,12 @@
 
 namespace System.Reflection.TypeExtensions.Tests.MemberInfo
```
Comment (stephentoub) [inline src/System.Reflection.TypeExtensions/tests/MemberInfoTests.cs]: Should the namespace just be System.Reflection.Tests?

File: src/System.Reflection.TypeExtensions/tests/MemberInfo/MemberInfoMemberType.cs (hunk 1 above)
Comment (hughbe) [inline src/System.Reflection.TypeExtensions/tests/MemberInfo/MemberInfoMemberType.cs]: Sorry, should have mentioned.
They exist [here](https://github.com/dotnet/corefx/blob/master/src/System.Reflection.TypeExtensions/tests/FieldInfoTests.cs#L15) and [here](https://github.com/dotnet/corefx/blob/master/src/System.Reflection.TypeExtensions/tests/FieldInfoTests.cs#L20)

File: src/System.Reflection.TypeExtensions/tests/MemberInfo/MemberInfoMemberType.cs (hunk 2 above)
Comment (hughbe) [inline src/System.Reflection.TypeExtensions/tests/MemberInfo/MemberInfoMemberType.cs]: I've added a test for `IsNestedType` in the PR feedback commit.

File: src/System.Reflection.TypeExtensions/tests/MemberInfoTests.cs (hunk 3 above)
Comment (hughbe) [inline src/System.Reflection.TypeExtensions/tests/MemberInfoTests.cs]: Whoops, copy-paste. Fixed

File: src/System.Reflection.TypeExtensions/tests/MemberInfoTests.cs (hunk 4 above)
Comment (hughbe) [inline src/System.Reflection.TypeExtensions/tests/MemberInfoTests.cs]: Correct, fixed

**Comment by stephentoub** [comment]
Thanks. LGTM.

**Pull request merged**